### PR TITLE
Ticket 5.

### PR DESCRIPTION
**Pull request by johnnywalker-git**
Added endpoint to retrieve all articles.
Returns an object with an array of articles sorted in descending order by date.
Also adds number of comments for each article.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -78,4 +78,33 @@ describe("GET API/articles/:article_id", () => {
...
+        expect(article).not.toHaveProperty("body")
+      })
+    })
+  })
+  test("200: First item in returned array should be the the least recent article" ,() => {
+    return request(app)
+    .get("/api/articles")
+    .expect(200)
+    .then(({body}) => {
+      expect([body[0].created_at, body[1].created_at]).toBeSorted({ descending: true })
```
Comment (Dutton1990) [inline __tests__/app.test.js:107]: If you wanted to make this a little neater toBeSortedBy allows you to pass an argument of what you want the data to be sorted by - but this works!

File: models/articles.model.js [hunk 2]
```diff
@@ -13,4 +13,20 @@ exports.readArticle = (article) => {
...
+        return db.query
+   (`UPDATE articles SET comments_count = (SELECT COUNT(*) FROM comments WHERE comments.article_id = articles.article_id);`)
+    })
+    .then(() => {
+        return db.query
+   (`SELECT author,title,article_id,topic,created_at,votes,article_img_url,comments_count FROM articles ORDER BY created_at DESC;`)
+    })
+    .then(({rows}) => {
+    return{"allArticles": rows}
+   })
```
Comment (Dutton1990) [inline models/articles.model.js:26]: This unfortunately is not quite what the ticket is asking for. Here you are directly altering the database - what we would like you to do is essentially do all of this inside a select. So the comment_count column is generated through a query and not added to the database. 

You have the code here basically - just see if you can get it working within the select on line 27 and without the alter table/update!

**Review by Dutton1990 — CHANGES_REQUESTED** [review]
Just need a change with that model Johnathan! Once you have that please resubmit so we can check it before you merge!

File: models/articles.model.js (hunk 2 above)
Comment (johnnywalker-git) [reply models/articles.model.js:26]: Does it need a join? Struggling to make it work!

**Review by NRMSMITH — APPROVED** [review]
It did need a join (sorry we didn't get back to you in time!!!!!), but well done for figuring it out, joins are serious business. I'm happy for you to merge now 🚀 ☄️

**Comment by johnnywalker-git** [comment]
Joins are srs 'annoying' business!! Thank you :D

**Pull request merged**